protocol/superc20/liquidity-migration.md

```diff
@@ -0,0 +1,60 @@
+# Liquidity Migration
+
+## Purpose
+
+_The document presented below is part of the [Interoperability project](https://github.com/ethereum-optimism/optimism/issues/10899)._
+
+This document proposes a solution for migrating the liquidity of tokens locked in L1, deployed through the current and legacy `OptimismMintableERC20` Factories in L2 (also known as _legacy tokens_), to make them fungible and compatible with Superchain interoperability.
+
+## Summary
+
+We must deploy new token representations because the current L2 tokens can't be upgraded to be compatible with Superchain interoperability. The main goal is to minimize any issues with liquidity migration.
+
+In this document, we will explore three approaches to create a representation that complies with `ISuperchainERC20`: Wrap, Convert, and Mirror. Among these, Mirror seems to be the best option as it doesn't force users to migrate to new token contracts.
+
+## Problem Statement + Context
```

Comment:
Does anyone have concrete data on the following things:

- TVL of immutable bridged tokens created with the `OptimismMintableERC20Factory`
- TVL of native L2 tokens
- TVL of bridged tokens that are upgradeable

This would be highly valuable IMO.

Reply:
TVL of bridged tokens that are upgradeable:
TVL of immutable bridged tokens created with the OptimismMintableERC20Factory: 
TVL of native L2 tokens: https://l2beat.com/scaling/projects/optimism/tvl-breakdown
TVL of L1StandardBridge: https://etherscan.io/address/0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1#multichain-portfolio

Reply:
One metapoint: Want to clarify when we mean TVL versus Assets Onchain

The L2Beat and Etherscan links to do show assets on chain, I can work on app TVL (assuming these are the assets that will actually move vs being stuck in CEXs / wallets).

---

Second, is the `bridged tokens that are upgradeable` / `immutable bridged tokens` distinction just if they're behind a proxy?

--

I also imagine we're talking about as many Superchain members as we can get, not just OP Mainnet right?

Reply:
@MSilb7 context of the question is: how much of assets bridged are upgradable / non-upgradable. If majority is non-upgradable then we need to prioritize mirroring, if its upgradable we might not need to pri mirroring for the initial release. 

Re members - starting with Base, OPM would be great as that's the majority of liquidity to inform the above. 

Proxy would work to determine upgradable!

Reply:
Ok yeah, if you all have a good way to identify which tokens are proxies vs not, you can match that up against L2Beat.

I know USDC is a proxy as an example. We could ~try to approximate it like "did the token ever emit an upgrade event."

Reply:
What contract(s) are the `OptimismMintableERC20Factory`?

I was looking at `0x4200000000000000000000000000000000000012` on OPM and Base, but I'm seeing some tokens on L2Beat that were minted on Base via `0xf23d369d7471bD9f6487E198723eEa023389f1d4`. Are the custom ERC20 factories, or are these custom ones not valid?

Reply:
per [Base docs](https://docs.base.org/docs/base-contracts/#base-mainnet), seems like this is another token factory `0xF10122D428B4bc8A9d050D06a2037259b4c4B83B`

If chains are rolling their own token factories, we'll likely need a registry of this.

Reply:
Looked at Assets Onchain for Base and OPM
- Proxy Tokens, Non-Proxy, Native (ETH) even about 25% each (~30% each excl. predeploys)
- Immutable token factory 8%
- OP (Predeploy) 14.7%

<img width="811" alt="image" src="https://github.com/ethereum-optimism/design-docs/assets/4006780/2367691a-5d81-47bf-b43a-dcdf6b986e22">


- Base rolled custom minting factories.
- "Unknown" = Token was deployed before regenesis, or does not have an onchain L2<>L1 mapping.
- Assets Onchain value sourced from L2Beat & Mapped to the L2 token address.
- Tokens created on OP Mainnet pre-regenesis were labeled as "Unknown," unless a proxy event was observed post-regenesis.

Data Issues:
- Some L1 tokens map to multiple L2 tokens by bridge events. In this case, we used amount of dex swaps as the indicator for which version of the token to treat as the adopted version.
- The immutable token factory events seem to be spoofable, which will cause token classification errors, if you go off of factory events (i.e. Native USDC appears to be [factory created](https://basescan.org/tx/0x04a19b13b48464557d394ab2cdcb7a578642a0e8d898ec4a632da92066b5acf4#eventlog), when it [is not](https://basescan.org/tx/0x8aa214f98bcf2984add809d10232135cccc4d6ab97d8477e66475d8bf68def34)). I manually overrode USDC, but there may be other bugs.
- ^ Request: **Can we make sure there's no ability for people to spoof token creations like this going forward?**